lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb

```diff
@@ -2,468 +2,344 @@
-            @meta_field = initialize_client(:MetaField)
-            @modifier = initialize_client(:Modifier)
-            @modifier_value = initialize_client(:ModifierValue)
-        end
+    def working?
+      received_event_without_error?
+    end
 
-        def initialize_client(class_name)
-            klass = ::BigcommerceProductAgent::Client.const_get(class_name.to_sym)
-            return klass.new(
-                interpolated['store_hash'],
-                interpolated['client_id'],
-                interpolated['access_token']
-            )
-        end
+    def check
+      initialize_clients
+      handle interpolated['payload'].presence || {}
+    end
 
-        def get_mapper(class_name)
-            return ::BigcommerceProductAgent::Mapper.const_get(class_name.to_sym)
-        end
+    def receive(incoming_events)
+      initialize_clients
+      incoming_events.each do |event|
+        handle(event)
+      end
+    end
 
-        def upsert_product(sku, product, bc_product = nil, is_digital=false, search_skus=[])
-            custom_fields_updates = get_mapper(:CustomFieldMapper).map(
-                interpolated['custom_fields_map'],
-                product,
-                bc_product
-            )
-
-            product_id = bc_product['id'] unless bc_product.nil?
-
-            payload = get_mapper(:ProductMapper).payload(
-                sku,
-                product,
-                product_id,
-                {
-                    additional_search_terms: search_skus,
-                    custom_fields: custom_fields_updates[:upsert]
-                },
-                is_digital,
-            )
-
-            payload[:is_visible] = false
-            # NOTE:  Products are intentionally upserted as disabled so that users
-            # don't see an incomplete listing (particularly when leveraging variants)
-            # The variant and option_list methods will re-enable products after the
-            # upsert is complete.
-
-            bc_product = @product.upsert(payload, {
-                include: %w[custom_fields variants options].join(',')
-            })
-
-            # Metafields need to be managed separately. Intentionally get them _AFTER_
-            # the upsert so that we have the necessary resource_id (bc_product.id)
-            meta_fields_updates = get_mapper(:MetaFieldMapper).map(
-                interpolated['meta_fields_map'],
-                product,
-                bc_product,
-                @meta_field.get_for_product(bc_product['id']),
-                interpolated['meta_fields_namespace']
-            )
-
-            {
-                product: bc_product,
-                custom_fields_delete: custom_fields_updates[:delete],
-                meta_fields_upsert: meta_fields_updates[:upsert],
-                meta_fields_delete: meta_fields_updates[:delete]
-            }
-        end
+    def handle(event)
+      acumen_products = event.payload
```

Comment:
@abourge I had a bit of a brain fart on the naming conventions. We actually want the agents to be client agnostic.

Can we change uses of `acumen_product` to something like `input_product` or `raw_product`?